Here is a stacktrace that shows the issue

```
INTERNAL_SERVER_ERROR: com.google.cloud.storage.StorageException: java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "setFactory")
        at com.google.cloud.storage.StorageException.translateAndThrow(StorageException.java:74)
        at com.google.cloud.storage.BlobWriteChannel.flushBuffer(BlobWriteChannel.java:67)
        at com.google.cloud.BaseWriteChannel.flush(BaseWriteChannel.java:112)
        at com.google.cloud.BaseWriteChannel.write(BaseWriteChannel.java:139)
        at java.base/java.nio.channels.Channels.writeFullyImpl(Channels.java:74)
        at java.base/java.nio.channels.Channels.writeFully(Channels.java:97)
        at java.base/java.nio.channels.Channels$1.write(Channels.java:172)
        at java.base/java.io.InputStream.transferTo(InputStream.java:777)
        at org.elasticsearch.core.internal.io.Streams.copy(Streams.java:47) <-- this does not have the necessary permission
        at org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStore.lambda$writeBlobResumable$5(GoogleCloudStorageBlobStore.java:216)
        at org.elasticsearch.repositories.gcs.SocketAccess.lambda$doPrivilegedVoidIOException$0(SocketAccess.java:54)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:554)
        at org.elasticsearch.repositories.gcs.SocketAccess.doPrivilegedVoidIOException(SocketAccess.java:53)
        at org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStore.writeBlobResumable(GoogleCloudStorageBlobStore.java:215)
        at org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobStore.writeBlob(GoogleCloudStorageBlobStore.java:186)
        at org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobContainer.writeBlob(GoogleCloudStorageBlobContainer.java:67)
        at org.elasticsearch.repositories.blobstore.BlobStoreRepository.snapshotFile(BlobStoreRepository.java:2075)
        at org.elasticsearch.repositories.blobstore.BlobStoreRepository.lambda$snapshotShard$55(BlobStoreRepository.java:1745)
        at org.elasticsearch.action.ActionRunnable$1.doRun(ActionRunnable.java:45)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at java.base/java.lang.Thread.run(Thread.java:832)
        Suppressed: com.google.cloud.storage.StorageException: java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "setFactory")
                at com.google.cloud.storage.StorageException.translateAndThrow(StorageException.java:74)
                at com.google.cloud.storage.BlobWriteChannel.flushBuffer(BlobWriteChannel.java:67)
                at com.google.cloud.BaseWriteChannel.close(BaseWriteChannel.java:151)
                at java.base/java.nio.channels.Channels$1.close(Channels.java:177)
                at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:104)
                at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:74)
                at org.elasticsearch.core.internal.io.Streams.copy(Streams.java:54)
                ... 15 more
        Caused by: java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "setFactory")
                at java.base/java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
                at java.base/java.security.AccessController.checkPermission(AccessController.java:1036)
                at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:408)
                at java.base/java.lang.SecurityManager.checkSetFactory(SecurityManager.java:1487)
                at java.base/javax.net.ssl.HttpsURLConnection.setSSLSocketFactory(HttpsURLConnection.java:365)
                at com.google.api.client.http.javanet.NetHttpTransport.buildRequest(NetHttpTransport.java:158)
                at com.google.api.client.http.javanet.NetHttpTransport.buildRequest(NetHttpTransport.java:55)
                at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:889)
                at com.google.cloud.storage.spi.v1.HttpStorageRpc.write(HttpStorageRpc.java:753)
                at com.google.cloud.storage.BlobWriteChannel$1.run(BlobWriteChannel.java:60)
                at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
                at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
                at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
                at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
                at com.google.cloud.storage.BlobWriteChannel.flushBuffer(BlobWriteChannel.java:53)
                ... 20 more
```

--- a/plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java
+++ b/plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java
@@ -34,6 +34,7 @@ import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.message.ParameterizedMessage;
 import org.elasticsearch.ExceptionsHelper;
+import org.elasticsearch.common.SuppressForbidden;
 import org.elasticsearch.common.blobstore.BlobContainer;
 import org.elasticsearch.common.blobstore.BlobMetadata;
 import org.elasticsearch.common.blobstore.BlobPath;
@@ -246,6 +247,7 @@ class GoogleCloudStorageBlobStore implements BlobStore {
                  */
                 Streams.copy(inputStream, Channels.newOutputStream(new WritableByteChannel() {
 
+                    @SuppressForbidden(reason = "channel is based on a socket")
                     @Override
                     public int write(final ByteBuffer src) throws IOException {
                         return SocketAccess.doPrivilegedIOException(() -> writeChannel.write(src));

--- a/plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java
+++ b/plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java
@@ -82,9 +82,6 @@ class GoogleCloudStorageBlobStore implements BlobStore {
         if (largeBlobThresholdByteSizeProperty == null) {
             LARGE_BLOB_THRESHOLD_BYTE_SIZE = Math.toIntExact(new ByteSizeValue(5, ByteSizeUnit.MB).getBytes());
         } else {
-            if (true) {
-                throw new IllegalArgumentException();
-            }
             final int largeBlobThresholdByteSize;
             try {
               largeBlobThresholdByteSize = Integer.parseInt(largeBlobThresholdByteSizeProperty);

